File: 1-Part/4-chapter/postorderTraversal.py
root = {
    'data': 'A', 'children': [
        {'data': 'B', 'children': [
            {'data': 'D', 'children': []}
        ]},
        {'data': 'C', 'children': [
            {'data': 'E', 'children': [
                {'data': 'G', 'children': []},
                {'data': 'H', 'children': []}
            ]},
            {'data': 'F', 'children': []}
        ]}
    ]
}


def postorderTraverse(node):
    for child in node['children']:
        postorderTraverse(child)
    print(node['data'], end=' ')

    return


postorderTraverse(root)

File: 1-Part/4-chapter/postorderTraversal.js
let root = {
    'data': 'A', 'children': [
        {
            'data': 'B', 'children': [
                {'data': 'D', 'children': []}
            ]
        },
        {
            'data': 'C', 'children': [
                {
                    'data': 'E', 'children': [
                        {'data': 'G', 'children': []},
                        {'data': 'H', 'children': []}
                    ]
                },
                {'data': 'F', 'children': []}
            ]
        }
    ]
}

function postorderTraverse(node) {
    for (let i = 0; i < node['children'].length; i++) {
        postorderTraverse(node['children'][i])
    }
    console.log(node['data'])

    return
}

postorderTraverse(root)